source/agora/node/Ledger.d

```diff
@@ -229,6 +233,84 @@ public class Ledger
         else
             return null;
     }
+
+    /***************************************************************************
+
+        Write the block on the ledgerFile.
+
+        Params:
+            T = Type of struct to writeFile
+            record = Instance of `T` to serialize
+            dg  = Serialization delegate, when this struct is a nested struct
+
+    ***************************************************************************/
+
+    public void writeBlockFile (T) (auto ref T record) @safe
+    if (is(T == struct))
+    {
+        auto fd = File("ledger.dat", "w"); // TODO move to config.yaml
+        scope SerializeDg dg = (scope const (ubyte[]) data) @safe
+        {
+            try
+            {
+                fd.rawWrite(data);
+            }
+            catch (ErrnoException ex)
+            {
+                //logError("writeBlockFile: %s", ex.message);
+            }
+        };
+        serializePart(record, dg);
+    }
+
+    /***************************************************************************
+
+        Return raw uytes read in the ledger to uyte[].
+
+        Params:
+            dg  = Deserialization delegate, when this struct is a nested struct
+
+        Return:
+            Return Block[]
+
+    ***************************************************************************/
+
+    public Block[] readLedgerFile () @safe
+    {
+        scope ubyte[] data;
+        try
+        {
+            ubyte[] buffer;
+            buffer.length = 1024;
+            auto file = File("ledger.dat", "r"); // TODO move to config.yaml
+            data = file.rawRead(buffer);
+        }
+        catch (ErrnoException ex)
+        {
+            //logError("readLedgerFile: %s", ex.message);
```

Comment:
You are ignoring the exception here.

Then in the delegate `dg` below, you will try to slice `data[offset .. offset + size];`, which will throw another exception because `data` is empty and you try to index more data than exists.

This will make the node shut down on startup when it tries to read the blocks.